src/main/java/org/traccar/protocol/TotemProtocolEncoder.java

```diff
@@ -16,40 +16,27 @@
  */
 package org.traccar.protocol;
 
-import org.traccar.StringProtocolEncoder;
 import org.traccar.model.Command;
 import org.traccar.Protocol;
+import org.traccar.helper.Checksum;
 
-public class TotemProtocolEncoder extends StringProtocolEncoder {
+public class TotemProtocolEncoder extends TotemProtocolSmsEncoder {
```

Comment:
I don't think it makes sense to extend it. One possible option is to define a static function here that can be reused. I also would recommend defining it here in `TotemProtocolEncoder`.

Reply:
The problem with static function is that I cannot use formatCommand, as it's non-static and I would like to avoid duplicating it. Any ideas?

Reply:
That's why I said you can define (a new) static function and reuse it in both places.

Reply:
Sorry I'm not sure I follow, if I move the `getCommandString` to TotemProtocolEncoder and declare it as a static function (to reuse it in TotemProtocolSmsEncoder) I get `error: non-static method formatCommand(Command,String,String...) cannot be referenced from a static context`.

Sorry if this is basic Java, I'm more home in non strongly typed languages :D

Reply:
The idea is to do something like this:
```java
class TotemProtocolEncoder {
    static String formatContent(...) { ... }
    Object formatCommand(...) { ... formatContent(...) ... }
}

class TotemProtocolSmsEncoder {
    Object formatCommand(...) { ... TotemProtocolEncoder.formatContent(...) ... }
}
```